src/saga-dense.h

```diff
@@ -129,8 +131,9 @@ Saga(Penalty&               penalty,
   double wscale = 1.0;
 
   double wscale_update = 1.0 - alpha*gamma;
+  if (noncov > 0.0) { wscale_update = 1.0; }
```

Comment:
What happens if the user uses the elastic net penalty but still changes the non-convexity parameter? I might be wrong, but it seems like it will then mess up the results for elastic net. If I'm right, please consider always setting noncov = 1 if using the elastic net penalty (ignoring the input to sgdnet()).

I also suggest this formatting change:

```suggestion
  if (noncov > 0.0)
    wscale_update = 1.0;
```

Reply:
I change the default noncov, the previous default value is not good. I make use of existing structure of computing alpha and beta for mnet/snet (mcp/scad + ridge). The algorithm will still use elastic net penalty if user do not set a legal non-convexity parameter now.